### PR TITLE
chore(lint): add ESLint to repo

### LINKED ISSUE #64: Replace JSHint with ESLint
... just like all the other fxa-* repos.

### PR DESCRIPTION
**Pull request by TDA**
Fixes #64
Remove JSHint
Add ESLint
Fix linting errors
@pdehaan

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -3,6 +3,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 module.exports = function (grunt) {
+  'use strict';
```
Comment (TDA) [inline Gruntfile.js]: Do we need to use strict here @pdehaan ? seeing as this is node code.

File: node/amd-loader.js [hunk 2]
```diff
@@ -7,6 +7,7 @@
 var path = require('path');
 
 module.exports = function amdload(absoluteFilename, baseUrl, map) {
+  'use strict';
```
Comment (TDA) [inline node/amd-loader.js]: Same. use strict in node? @pdehaan

File: client/lib/options.js [hunk 3]
```diff
@@ -23,7 +23,7 @@ define([], function () {
    * if a required option is missing
    */
   function checkRequired(requiredOptions, options) {
-    for (var i = 0, requiredOption; requiredOption = requiredOptions[i]; ++i) {
+    for (var i = 0, requiredOption; requiredOption = requiredOptions[i]; ++i) { //eslint-disable-line no-cond-assign
```
Comment (TDA) [inline client/lib/options.js]: It looks like we need to assign here. So i disabled lint, is that right? @pdehaan

File: client/lib/object.js [hunk 4]
```diff
@@ -20,7 +20,7 @@ define([], function () {
   function extend(target/*, ...*/) {
     var sources = [].slice.call(arguments, 1);
 
-    for (var index = 0, source; source = sources[index]; ++index) {
+    for (var index = 0, source; source = sources[index]; ++index) { //eslint-disable-line no-cond-assign
```
Comment (TDA) [inline client/lib/object.js]: Same as before. @pdehaan

File: package.json [hunk 5]
```diff
@@ -29,18 +29,19 @@
   },
   "homepage": "https://github.com/mozilla/fxa-relier-client",
   "devDependencies": {
+    "eslint-config-fxa": "1.6.0",
```
Comment (pdehaan) [inline package.json]: Bump this to 1.7.0.

Comment (TDA) [inline package.json]: Done - da - done!

**Comment by vladikoff** [comment]
@TDA remove `use strict` from node code (and Grunt tasks).

**Comment by TDA** [comment]
@vladikoff Will do! :)

**Comment by TDA** [comment]
@vladikoff Travis is greeen :dancer:

**Pull request merged**